ajax/loadfile.php

```diff
@@ -30,25 +30,31 @@
 // Set the session key for the file we are about to edit.
 $dir = isset($_GET['dir']) ? $_GET['dir'] : '';
 $filename = isset($_GET['file']) ? $_GET['file'] : '';
-if(!empty($filename))
-{
-	$path = $dir.'/'.$filename;
-	$writeable = \OC\Files\Filesystem::isUpdatable($path);
-	$mime = \OC\Files\Filesystem::getMimeType($path);
-	$mtime = \OC\Files\Filesystem::filemtime($path);
-	$filecontents = \OC\Files\Filesystem::file_get_contents($path);
-	$encoding = mb_detect_encoding($filecontents."a", "UTF-8, WINDOWS-1252, ISO-8859-15, ISO-8859-1, ASCII", true);
-	if ($encoding == "") {
-		// set default encoding if it couldn't be detected
-		$encoding = 'ISO-8859-15';
+
+try {
+
+	if (!empty($filename)) {
+		$path = $dir . '/' . $filename;
+		$writeable = \OC\Files\Filesystem::isUpdatable($path);
+		$mime = \OC\Files\Filesystem::getMimeType($path);
+		$mtime = \OC\Files\Filesystem::filemtime($path);
+		$filecontents = \OC\Files\Filesystem::file_get_contents($path);
+		$encoding = mb_detect_encoding($filecontents . "a", "UTF-8, WINDOWS-1252, ISO-8859-15, ISO-8859-1, ASCII", true);
+		if ($encoding == "") {
+			// set default encoding if it couldn't be detected
+			$encoding = 'ISO-8859-15';
+		}
+		$filecontents = iconv($encoding, "UTF-8", $filecontents);
+		OCP\JSON::success(array('data' => array(
+				'filecontents' => $filecontents,
+				'writeable' => $writeable,
+				'mime' => $mime,
+				'mtime' => $mtime))
+		);
+	} else {
+		OCP\JSON::error(array('data' => array('message' => 'Invalid file path supplied.')));
 	}
-	$filecontents = iconv($encoding, "UTF-8", $filecontents);
-	OCP\JSON::success(array('data' => array(
-		'filecontents' => $filecontents,
-		'writeable' => $writeable,
-		'mime' => $mime,
-		'mtime' => $mtime))
-	);
-} else {
-	OCP\JSON::error(array('data' => array( 'message' => 'Invalid file path supplied.')));
+
+} catch (\Exception $e) {
+	OCP\JSON::error(array('data' => array('message' => $e->getMessage())));
```

Comment:
> see - exception messages sometimes hit the user

Yes, this is what we discussed when we worked in encryption 2.0. Do you remember? We decided to remove all the hard-coded encryption errors outside of the encryption app and agreed that instead the encryption app should throw useful errors which can then be catched and displayed to the user.